repeat_expansion_variants/clinvar_identifier_parsing.py

```diff
@@ -0,0 +1,108 @@
+    r':'                                 # Delimiter, transcript/variant info        :
+)
+
+# Common part for start and end coordinate pivots. Pivots for introns and other cases where a coordinate in a noncoding
+# region of mRNA needs to be addressed relative to the coding regions. For example, c.87 means the 87th coding base of
+# mRNA (not counting introns). In comparison, c.87+14 means that base number 87 is the last base of a particular exon,
+# and the base addressed by the coordinate is 14 bases downstream of the pivot base. In this case, an intron repeat
+# expansion variant might be addressed as c.87+14_c.87+17. In this case, we don't want the pivots (87), but only the
+# actual coordinates (+14 and +17). To do that, we have a regular expression which captures the pivot part.
+coordinate_pivot_part = (
+    r'(?:'       # Non-capturing group for coordinate pivot part
+    r'[-+]?'     # Coordinate pivot may start with a minus or plus
+    r'[0-9]+'    # Then it contains a number
+    r'(?=[-+])'  # It then must be followed with either a plus or minus.  This is a positive lookahead and is *not*
+                 # part of the coordinate pivot, hence the (?=...) notation
+    r')?'        # The coordinate pivot is always optional
+)
+
+# Pattern 1. HGVS-like notation for coding or genomic coordinates. This pattern is used for most variants. From it we
+# can always extract transcript ID and start coord, and sometimes also end coord and repeat unit sequence.
+# Example: 'NM_001256054.2(C9orf72):c.-45+63_-45+80GGGGCC(2_25)'
+re_hgvs_like_transcript_or_genomic = re.compile(
+    hgvs_like_transcript_part +      # Transcript definition                                NM_001256054.2(C9orf72):
+    r'[gc]'                          # Coordinate type, genomic or coding                   c
+    r'\.'                            # Delimiter, coordinate type/coordinate                .
+
+    + coordinate_pivot_part +        # Start coordinate pivot, optional                     -45
+    r'\*?'                           # Sometimes there is an asterisk in front of the
+                                     # coordinate (special case, not important)
+    r'(?P<start_coord>[+-]?[0-9]+)'  # Start coordinate                                     +63
+
+    r'(?:'                           # Non-capruting group for end coordinate
+    r'_'                                 # Delimiter: start coordinate/end coordinate       _
+    + coordinate_pivot_part +            # End coordinate pivot, optional                   -45
+    r'\*?'                               # Coordinate may start with an asterisk
+    r'(?P<end_coord>[+-]?[0-9]+)'        # End coordinate                                   +80
+    r')?'                            # The entire end coordinate part is optional
+
+    r'(?P<sequence>[ATGC]*)'         # Repeat unit sequence, optional                       GGGGCC
```

Comment:
Guessing there is no need for IUPAC set of nucleotides here?

Reply:
Currently, ClinVar data only contains [ATGC] and sometimes an N. (N was not a part of the original regexp; I noticed that later and fixed it in the next PR, #4.)

However, I think I'm going to lean on the safe side here and include the IUPAC symbols as well. I can see no harm in doing so. Fixed in https://github.com/EBIvariation/vep-mapping-pipeline/pull/3/commits/f4d93cd0853e3363179d2c6752a93ca6c1f6142b.